doc/usage/extensions/autodoc.rst

```diff
@@ -546,3 +555,36 @@ member should be included in the documentation by using the following event:
       ``inherited_members``, ``undoc_members``, ``show_inheritance`` and
       ``noindex`` that are true if the flag option of same name was given to the
       auto directive
+
+Removing Type Hints
+-------------------
+
+.. versionadded:: 2.1
+
+The ``documenter`` parameter to :event:`autodoc-process-signature` enables users
+to remove type hints from docstring signatures if desired.  In ``conf.py``::
+
+    def autodoc_process_signature(app, what, name, obj, options, signature,
+                                  return_annotation, documenter):
+        """Remove type hints from all documented items."""
+        try:
+            return documenter.format_args(show_annotation=False), return_annotation
+        except:
+            return None  # signals no change
+
+
+    def setup(app):
```

Comment:
Are you saying almost people want to control removal of typehints manually instead of configuration? I don't think so. I also feel the controllable pruning would not be needed for many people.
Do you want to switch the pruning behavior for individual functions or methods? It is not consistent document to me.